src/wagtail_localize_smartling/viewsets.py

```diff
@@ -0,0 +1,206 @@
+import django_filters
+
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from django.utils.functional import cached_property
+from django.utils.html import format_html, format_html_join
+from django.utils.translation import gettext_lazy as _
+from wagtail.admin.filters import WagtailFilterSet
+from wagtail.admin.ui.tables import Column, DateColumn, TitleColumn, UserColumn
+from wagtail.admin.views import generic
+from wagtail.admin.viewsets.model import ModelViewSet
+from wagtail.permission_policies import ModelPermissionPolicy
+
+from .api.types import JobStatus
+from .models import Job
+from .templatetags.wagtail_localize_smartling_admin_tags import smartling_job_url
+
+
+def get_users_for_filter(user):
+    User = get_user_model()
+    return User.objects.filter(
+        pk__in=Job.objects.values_list("user", flat=True)
+    ).order_by(User.USERNAME_FIELD)  # pyright: ignore[reportAttributeAccessIssue]
+
+
+class JobReportFilterSet(WagtailFilterSet):
+    status = django_filters.ChoiceFilter(choices=JobStatus.choices)
+    user = django_filters.ModelChoiceFilter(
+        label=_("User"),
+        field_name="user",
+        queryset=lambda request: get_users_for_filter(request.user),
```

Comment:
Question: why does this take `request`? It looks like the `user` arg to `get_users_for_filter` is unused.

Reply:
https://django-filter.readthedocs.io/en/main/guide/usage.html#filtering-the-related-queryset-for-modelchoicefilter

> The queryset argument for ModelChoiceFilter and ModelMultipleChoiceFilter supports callable behavior. If a callable is passed, it will be invoked with the request as its only argument. This allows you to perform the same kinds of request-based filtering without resorting to overriding FilterSet.__init__.